janus_core/calculations/phonons.py

```diff
@@ -278,11 +284,27 @@ def __init__(
         self.temp_max = temp_max
         self.temp_step = temp_step
         self.force_consts_to_hdf5 = force_consts_to_hdf5
+        self.hdf5 = hdf5
         self.plot_to_file = plot_to_file
         self.write_results = write_results
         self.write_full = write_full
         self.enable_progress_bar = enable_progress_bar
 
+        # Handle deprecation
+        if force_consts_to_hdf5:
+            if hdf5:
```

Comment:
```suggestion
            if hdf5 is False:
```
Since the default for `hdf5` is `True`, I think we may want to check against it being set `False`, otherwise this triggers even if only `force_consts_to_hdf5` is set.

Alternatively, we could set the default `hdf5` to `None` initially, and set the actual default here, which would allow us to do the same `not None` check as above, and also catch setting `hdf5` to be `True`.